plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -491,7 +491,8 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
       bool process_incoming_transaction_async(const transaction_metadata_ptr& trx,
                                               bool persist_until_expired,
                                               next_function<transaction_trace_ptr> next,
-                                              RetryLaterFunc retry_later)
+                                              RetryLaterFunc retry_later,
+                                              bool return_failure_trace = false)
```

Comment:
This seems pretty fragile to me.  I think you'd be better of making sure that all downstream consumers can handle a transaction trace with an exception appropriately, and never extract the exception here.  Either that or somehow use types that enforce consistency between return_failure_trace and next.

Reply:
For now this will only be true for `amqp_trx_plugin`. However, a new `send_transaction2` in the future should also work this way (at least optionally). The community has requested this type of behavior.